mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1244,11 +1308,26 @@ impl TakerSwap {
 
         Ok((Some(TakerSwapCommand::WaitForTakerPaymentSpend), vec![
             TakerSwapEvent::TakerPaymentSent(tx_ident),
+            TakerSwapEvent::WatcherMessageSent(preimage_hex),
         ]))
     }
 
     async fn wait_for_taker_payment_spend(&self) -> Result<(Option<TakerSwapCommand>, Vec<TakerSwapEvent>), String> {
         let tx_hex = self.r().taker_payment.as_ref().unwrap().tx_hex.0.clone();
+        let preimage_hex = self.r().taker_spends_maker_payment_preimage.clone();
+        let mut watcher_broadcast_abort_handle = None;
+        if self.ctx.use_watchers() && preimage_hex.is_some() {
+            let watcher_data = self.create_watcher_data(tx_hex.clone(), preimage_hex.unwrap());
```

Comment:
Please consider using pattern matching if there are multiple conditions:
```rust
match preimage_hex {
  Some(preimage_hex) if self.ctx.use_watchers() => ...
  _ => (),
}
```
Or the nested if conditions:
```rust
if let Some(preimage_hex) = preimage_hex {
  if self.ctx.use_watchers() {
    ...
  }
}
```